compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1113,7 +1113,7 @@ class Namer { typer: Typer =>
 
       // println(s"final inherited for $sym: ${inherited.toString}") !!!
       // println(s"owner = ${sym.owner}, decls = ${sym.owner.info.decls.show}")
-      def isInline = sym.is(FinalOrInline, butNot = Method | Mutable)
+      def isInline = sym.is(FinalOrInlineOrTransparent, butNot = Method | Mutable)
```

Comment:
Not sure if it's worth adding `Transparent` to the exception here, since the handling of return types of transparent methods in `widenRhs` will have to be more different anyways (it should be allowed to infer a TermRef, for instance; think `transparent def unapply(x: Foo) = x`)

Reply:
Yes, that was only done as a temporary measure because I could not decide yet whether Java final constants should be called `inline(d)` or `transparent`. Probably `transparent`, but for now it's `inline`.